compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml

```diff
@@ -196,16 +196,25 @@ class HasFromAnyContractKey t k | t -> k where
   -- | HIDE
   _fromAnyContractKey : proxy t -> Any -> Optional k
 
+-- | Constraint satisfied by choices from templates with key.
+type ChoiceByKey t k c r =
```

Comment:
What do we need this for? You don’t seem to use this in any constraint afaict.

Reply:
We shouldn't use it in the stdlib. I think this might get used in script though, to update exerciseByKeyCmd?

Reply:
Maybe we can define it in `daml-script` then?

Reply:
All the other constraint tuples are defined here though, like `Template` and `TemplateKey` and `Choice`. 🤷‍♀️

Reply:
Fair point. But I think for daml script we actually want to be as relaxed a possible here. If we do use this one in Daml script it wouldn’t be possible to call `exerciseByKeyCmd` on a data-dependency from a previous template.

I’d probably suggest to just drop it until we start using it somewhere.

Reply:
That makes sense. 👍

Reply:
5f6f703a635f905f6a8d0226941bc756c1f1109f